File: scala-hoots/src/main/java/io/hoots/input/domain/Signature.java
package io.hoots.input.domain;

import javax.sound.sampled.AudioFormat;
import java.io.ByteArrayOutputStream;

/**
 * Created by rwadowski on 14.04.17.
 */
public class Signature {

    private final Item item;
    private final ByteArrayOutputStream stream;
    private final AudioFormat inputFormat;

    public Signature(Item item, ByteArrayOutputStream stream, AudioFormat inputFormat) {
        this.item = item;
        this.stream = stream;
        this.inputFormat = inputFormat;
    }

    public Item getItem() {
        return this.item;
    }

    public ByteArrayOutputStream getStream() {
        return this.stream;
    }

    public AudioFormat getInputFormat() {
        return this.inputFormat;
    }
}

File: scala-hoots/src/main/scala/io/hoots/input/domain/Signature.scala
package io.hoots.input.domain

import java.io.ByteArrayOutputStream
import javax.sound.sampled.AudioFormat

/**
  * Created by rwadowski on 19.04.17.
  */
case class Signature(item: Item,
                     byteArrayOutputStream: ByteArrayOutputStream,
                     audioFormat: AudioFormat)
